packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -1,244 +1,60 @@
-      const position = calculateSteps(ev, railRef, min, max, step, vertical, dir);
-      const currentStepPosition = Math.round(position / step) * step;
-
-      setRenderedPosition(position);
-      updateValue(currentStepPosition, ev);
-    },
-    [dir, max, min, step, updateValue, vertical],
-  );
-
-  const onPointerUp = React.useCallback(
-    (ev: React.PointerEvent<HTMLDivElement>): void => {
-      disposables.current.forEach(dispose => dispose());
-      disposables.current = [];
-      showStepAnimation();
-      // When undefined, the position fallbacks to the currentValue state.
-      setRenderedPosition(undefined);
-      inputRef.current!.focus();
-    },
-    [showStepAnimation],
-  );
-
-  const onPointerDown = React.useCallback(
-    (ev: React.PointerEvent<HTMLDivElement>): void => {
-      const { pointerId } = ev;
-      const target = ev.target as HTMLElement;
-
-      target.setPointerCapture?.(pointerId);
-
-      hideStepAnimation();
-      onPointerDownCallback?.(ev);
-
-      // eslint-disable-next-line deprecation/deprecation -- Should be remove an replaced with a useEvent hook.
-      disposables.current.push(on(target, 'pointermove', onPointerMove), on(target, 'pointerup', onPointerUp), () => {
-        target.releasePointerCapture?.(pointerId);
-      });
-
-      onPointerMove(ev);
-    },
-    [hideStepAnimation, onPointerDownCallback, onPointerMove, onPointerUp],
-  );
-
-  const onKeyDown = React.useCallback(
-    (ev: React.KeyboardEvent<HTMLDivElement>): void => {
-      hideStepAnimation();
-      onKeyDownCallback?.(ev);
-      const incomingValue = getKeydownValue(ev, currentValue, min, max, dir, keyboardStep);
-
-      if (currentValue !== incomingValue) {
-        updatePosition(incomingValue, ev);
-      }
-    },
-    [currentValue, dir, hideStepAnimation, keyboardStep, max, min, onKeyDownCallback, updatePosition],
-  );
-
-  const getTrackBorderRadius = () => {
-    if (origin !== undefined && origin !== (max || min)) {
-      if (vertical) {
-        return originPercent > valuePercent ? '99px 99px 0px 0px' : '0px 0px 99px 99px';
-      } else {
-        return (dir === 'rtl' ? valuePercent > originPercent : valuePercent < originPercent)
-          ? '99px 0px 0px 99px'
-          : '0px 99px 99px 0px';
-      }
-    }
-    return '99px';
-  };
-
-  useUnmount(() => {
-    disposables.current.forEach(dispose => dispose());
-    disposables.current = [];
-  });
-
-  const valuePercent = getPercent(renderedPosition !== undefined ? renderedPosition : currentValue, min, max);
-
+  const valuePercent = getPercent(currentValue, min, max);
   const originPercent = React.useMemo(() => {
     return origin !== undefined ? getPercent(origin, min, max) : 0;
   }, [max, min, origin]);
 
-  const markValues = React.useMemo((): number[] => getMarkValue(marks, min, max, step), [marks, max, min, step]);
+  const onChange: React.ChangeEventHandler<HTMLInputElement> = ev => {
```

Comment:
Use `useEventCallback` to avoid creating a new callback function every render.
```suggestion
  const onChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(ev => {
```

Reply:
To clarify, a new callback function will be created every render, but `useEventCallback` will return a wrapper function that's stable between renders, which calls whatever you pass in. I suppose it's a nuance; but just in case you were worried about stale values being captured in the lambda, that's not a problem with `useEventCallback`.